jack.py

```diff
@@ -739,12 +739,17 @@ def set_process_callback(self, callback, userdata=None):
             is called.
 
         """
-        @self._callback("JackProcessCallback", error=STOP_CALLING)
-        def callback_wrapper(frames, _):
-            return callback(frames, userdata)
+        if isinstance(callback, _ffi.CData):
+            callback_wrapper = callback
+            arg = _ffi.cast("void *", userdata)
```

Comment:
I would move the cast to `"void*"` out of here into the user code.
This way it would be more symmetrical: The user passes a `void*` and gets the same `void*` back within the callback function.

We should probably rename `arg` to `userdata_ptr` to make it a little clearer.
Then we can use

    callback_wrapper, userdata_ptr = callback, userdata

... here, and `userdata_ptr = _ffi.NULL` below.

What do you think?

At some point we might also consider renaming `callback_wrapper` to `callback_ptr` in the whole codebase, but I'm not yet sure if that's a good idea.